lib/src/pub.dart

```diff
@@ -29,11 +28,11 @@ const _flutterPackages = [
 
 /// Each of these is expensive to calculate; they require reading from disk.
 /// None of them changes during execution.
-/*late*/ Map<String, String> _nullSafePackageVersions;
-/*late*/ Map<String, String> _preNullSafePackageVersions;
+Map<String, String>? _nullSafePackageVersions;
+Map<String, String>? _preNullSafePackageVersions;
```

Comment:
Could you instead make this lazily initialized and then simplify the `getPackageVersions` method?

```suggestion
late final Map<String, String> _nullSafePackageVersions = packageVersionsFromPubspecLock(project.flutterTemplateProject(true));
late final Map<String, String> _preNullSafePackageVersions = packageVersionsFromPubspecLock(project.flutterTemplateProject(false));
```

Then turn the `getPackageVersions` method into the following:

```dart
/// Returns a mapping of Pub package name to package version.
Map<String, String> getPackageVersions({required bool nullSafe}) =>
    nullSafe ? _nullSafePackageVersions : _preNullSafePackageVersions;
```

Reply:
I've gone with your defaulting to null safety approach. Much betterer =)